### ISSUE TITLE
Unresolved conflict on servlet class: play.core.server.servlet30.Play2Servlet (JBoss 7.1.0)

### ISSUE THREAD

**Issue by peshkira** [play2war/play2-war-plugin#98]
I am trying to deploy a play 2.0.4 app on JBoss 7.1.0.
After following your guide, i get the following error:

```
11:08:06,272 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-7) MSC00001: Failed to start service jboss.deployment.unit."ROOT.war".PARSE: org.jboss.msc.service.StartException in service jboss.deployment.unit."ROOT.war".PARSE: Failed to process phase PARSE of deployment "ROOT.war"
    at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:119) [jboss-as-server-7.1.0.Final.jar:7.1.0.Final]
    at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1811) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
    at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1746) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
    at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886) [classes.jar:1.6.0_35]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908) [classes.jar:1.6.0_35]
    at java.lang.Thread.run(Thread.java:680) [classes.jar:1.6.0_35]
Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: JBAS018031: Conflict occurred processing web fragment in JAR: com.github.play2war.play2-war-core_2.9.1-0.5.jar
    at org.jboss.as.web.deployment.WarMetaDataProcessor.deploy(WarMetaDataProcessor.java:292)
    at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:113) [jboss-as-server-7.1.0.Final.jar:7.1.0.Final]
    ... 5 more
Caused by: java.lang.IllegalStateException: Unresolved conflict on servlet class: play.core.server.servlet30.Play2Servlet
    at org.jboss.metadata.merge.web.spec.ServletMetaDataMerger.augment(ServletMetaDataMerger.java:93)
    at org.jboss.metadata.merge.web.spec.ServletsMetaDataMerger.augment(ServletsMetaDataMerger.java:35)
    at org.jboss.metadata.merge.web.spec.WebCommonMetaDataMerger.augment(WebCommonMetaDataMerger.java:313)
    at org.jboss.as.web.deployment.WarMetaDataProcessor.deploy(WarMetaDataProcessor.java:290)
    ... 6 more

11:08:06,283 INFO  [org.jboss.as.server] (DeploymentScanner-threads - 2) JBAS015870: Deploy of deployment "ROOT.war" was rolled back with failure message {"JBAS014671: Failed services" => {"jboss.deployment.unit.\"ROOT.war\".PARSE" => "org.jboss.msc.service.StartException in service jboss.deployment.unit.\"ROOT.war\".PARSE: Failed to process phase PARSE of deployment \"ROOT.war\""}}
11:08:06,357 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) JBAS015877: Stopped deployment ROOT.war in 72ms
11:08:06,358 INFO  [org.jboss.as.controller] (DeploymentScanner-threads - 2) JBAS014774: Service status report
JBAS014777:   Services which failed to start:      service jboss.deployment.unit."ROOT.war".PARSE: org.jboss.msc.service.StartException in service jboss.deployment.unit."ROOT.war".PARSE: Failed to process phase PARSE of deployment "ROOT.war"

11:08:06,361 ERROR [org.jboss.as.server.deployment.scanner] (DeploymentScanner-threads - 1) {"JBAS014653: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-2" => {"JBAS014671: Failed services" => {"jboss.deployment.unit.\"ROOT.war\".PARSE" => "org.jboss.msc.service.StartException in service jboss.deployment.unit.\"ROOT.war\".PARSE: Failed to process phase PARSE of deployment \"ROOT.war\""}}}}
```

I am not 100% sure that the error is on your end, but I hope you can give me some insight.
Thanks!

[StackOverflow Question](http://stackoverflow.com/questions/12915199/deploy-play-2-app-to-jboss-7)

**Comment by dlecan** [comment]
Play2War version ?

According to this line (see the end of it) `Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: JBAS018031: Conflict occurred processing web fragment in JAR: com.github.play2war.play2-war-core_2.9.1-0.5.jar`, you are using Play2War v0.5, which is not compatible with Play 2.0.4.

Please upgrade to Play2war v0.7.4.

**Comment by peshkira** [comment]
Thanks for the quick reply, I just saw that too and updated.

The error is gone, but deployment still does not work. It just stops. Here is the jboss log. Is this related to issue #54?

```
11:23:00,757 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-3) JBAS015876: Starting deployment of "ROOT.war"
11:23:12,717 INFO  [org.apache.catalina.core.ContainerBase.[jboss.web].[default-host].[/]] (MSC service thread 1-7) PlayServletWrapper > contextInitialized
11:23:12,909 TRACE [org.jboss.modules] (MSC service thread 1-7) Finding class play.api.Logger$$anonfun$configure$1$$anonfun$apply$1 from Module "deployment.ROOT.war:main" from Service Module Loader
11:23:12,910 TRACE [org.jboss.modules] (MSC service thread 1-7) Finding local class play.api.Logger$$anonfun$configure$1$$anonfun$apply$1 from Module "deployment.ROOT.war:main" from Service Module Loader
11:23:12,910 TRACE [org.jboss.modules] (MSC service thread 1-7) Loading class play.api.Logger$$anonfun$configure$1$$anonfun$apply$1 locally from Module "deployment.ROOT.war:main" from Service Module Loader
11:23:12,911 TRACE [org.jboss.modules] (MSC service thread 1-7) Attempting to define class play.api.Logger$$anonfun$configure$1$$anonfun$apply$1 in Module "deployment.ROOT.war:main" from Service Module Loader
11:23:12,912 TRACE [org.jboss.modules] (MSC service thread 1-7) Defined class play.api.Logger$$anonfun$configure$1$$anonfun$apply$1 in Module "deployment.ROOT.war:main" from Service Module Loader
```

**Comment by dlecan** [comment]
> It just stops.
> Without #54, `PlayServletWrapper > contextInitialized` is the last lof you can see in the console from Play or your application. So you are not able to see if deployment is a success or a failure (and why)

Does `http://localhost:8080/` work ?

If not, something is wrong during the deployment, but you can't know what because of #54.
Play2War 0.8 is stable enough (including a fix for #54), I will release it soon.

**Comment by peshkira** [comment]
`http://localhost:8080/` serves an empty page.

Thanks for your reply. I will wait until 0.8.
cheers!

**Comment by dlecan** [comment]
Play2War 0.8 has been released, with a fix for issue #54.

**Comment by peshkira** [comment]
Works under 7.1.1. Logs are displayed accordingly.
I can't seem to be able to deploy under 7.1.0 but may be it is something on my end.

Thanks so much for your help! You rock! 
I consider this CLOSED as it solved the initial reported issue and more :).
